src/coreclr/vm/syncblk.cpp

```diff
@@ -2875,6 +2884,16 @@ BOOL SyncBlock::Wait(INT32 timeOut)
     GCPROTECT_END();
     m_Monitor.DecrementTransientPrecious();
 
+    if (isContentionKeywordEnabled)
```

Comment:
Just curious (not my area) given this is presumably a hot path, is there any advantage in moving the body of the if block into its own function? Or maybe PGO does this for us.

Reply:
It's already a very slow path. I haven't seen PGO do much outlining, but it may well move those blocks out of the way, which would be somewhat similar.